### ISSUE TITLE
how to dump node values(hidden unit values)

### ISSUE THREAD

**Issue by vidyatshetty** [Microsoft/CNTK#3179]
how to dump node value(hidden unit value) i.e, dense layer output using python?

**Comment by KeDengMS** [comment]
[set_computation_network_trace_level](https://cntk.ai/pythondocs/cntk.debugging.debug.html#cntk.debugging.debug.set_computation_network_trace_level) might help. Note that the output could be very verbose if the network is complicated.

**Comment by vidyatshetty** [comment]
@KeDengMS Thank you for writing. This was really helpful

**Comment by vidyatshetty** [comment]
@KeDengMS  I am using TIMIT database for speaker recognition. I am designing the system for 20 speakers. The TIMIT database has 10 samples for each speaker. so I am using 7 samples for training and 3 samples for testing. Is it possible that for each class I have very less samples and that is causing the overfitting of the model? However on testing with the training data I get 98% accuracy and on testing I get only 70%. I am struggling with this issue ffor a while.Please let me know

Thanks

**Comment by KeDengMS** [comment]
Seems overfitting to me. Some thoughts:
1. Apply some normalization on both training and test data, to make them have similar statistic distribution.
2. Or, try enhance the training data with more randomization, using truncation/scale/upsample/downsample etc.

**Comment by vidyatshetty** [comment]
@KeDengMS  Thank you for writing.
I will definitely work on the techniques you mentioned.

**Comment by KeDengMS** [comment]
Sorry I am not a data scientist, and I can only give you the common sense of fighting with overfitting. Please consult other data scientists who might have deeper insight in your problem.

Your model consists of only dense layer, which is know to learn more connections than necessary. For example, with MNIST dataset, if you shuffle all pixels of inputs in the same way so that human eyes can only see noise, dense layer would still able to train and converge which IMO is a kind of overfitting (since the trained network only works for that shuffled pattern).

**Comment by vidyatshetty** [comment]
@KeDengMS  Thank you for all the suggestions. This was really helpful

**Comment by vidyatshetty** [comment]
@KeDengMS  I used data augmentation and increased the number of samples per class from 9samples to 45samples per class. when I start to train the model the network doesn't learn anything after reaching the training loss = 1.9191 * 2048 metric = 72.69%. Is there anything I need to take care or change or modify the network when I increase the number of samples per class? Please let me know.
Thank you

**Comment by KeDengMS** [comment]
Closing as a dup of #3213.
